presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java

```diff
@@ -111,10 +115,24 @@ protected SchemaTableName getSchemaTableName(ResultSet resultSet)
     @Override
     protected String toSqlType(Type type)
     {
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            if (varcharType.getLength() <= MAX_IN_ROW_VARCHAR_LENGTH) {
```

Comment:
I undestand that 65k is shared among all columns. `toSqlType` is only used in one place: https://github.com/Teradata/presto/blob/feature_jdbc_varchar/presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java#L304. You can pass all column types there to detect case when sum of varchar lengths is greater then 65k. I such case you can just use `tinytext` type instead.